server.js

```diff
@@ -3,36 +3,260 @@ import bodyParser from 'body-parser'
+  },
+  isbn: {
+    type: String
+  },
+  isbn13: {
+    type: String
+  },
+  language_code: {
+    type: String,
+  },
+  num_pages: {
+    type: Number
+  },
+  ratings_count: {
+    type: Number
+  },
+  text_reviews_count: {
+    type: Number
+  }
+})
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Book.deleteMany()
+    booksData.forEach(async (book) => await new Book(book).save())
+  }
+  seedDatabase()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
+const myEndPoints = require('express-list-endpoints')
+const ERROR = { error: 'Data not found'}
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+//handling data connection error
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable'})
+  }
+})
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+    
+  
+  if(!res) {
+    res
+    .status(404)
+    .send({ error: 'Oops! Something goes wrong. Try again later!'})
+  }
+  res.send(myEndPoints(app))
+})
+//API endpoints for books with query options for author, title; otherwise, display 20 data per page (sorted by average_rating) and allow user to request for more pages
+app.get('/books', async (req, res) => {
+  const { title, author } = req.query
+
+  const queryTitleRegex = new RegExp(title, 'i')
+  const queryAuthorRegex = new RegExp(author, 'i')
+
+  if (title && author) {
+    const books = await Book.find({ title: queryTitleRegex, authors:queryAuthorRegex })
+    const returnObj = {
+      filterByTitle: title,
+      filterByAuthor: author,
+      amountOfData: books.length,
+      results: books
+    }
+    res.json(returnObj)
+  } else if (title) {
+    const books = await Book.find({ title: queryTitleRegex }).sort({ average_rating: -1 })
+    const returnObj = {
+      filterByTitle: title,
+      amountOfData: books.length,
+      results: books
+    }
+    res.json(returnObj)
+  } else if (author) {
+    const books = await Book.find({ authors: queryAuthorRegex }).sort({ average_rating: -1})
+    const returnObj = {
+      filterByAuthor: author,
+      amountOfData: books.length,
+      results: books
+    }
+    res.json(returnObj)
+  } else {
+    const { page } = req.query
+    const pageSize = 20
+    const skipData = page ? page * pageSize : 0;
+    
+    const amountOfData = await Book.find().count()
+    const totalNumberOfPage = Math.ceil(amountOfData / pageSize)
+    
+    let returnObj = {
+      totalAmountOfData: amountOfData,
+      totalNumberOfPage: totalNumberOfPage,
+      pageSize: pageSize,
+      currentPage: page || 1,
+      results: []
+    }
+
+    const books = await Book.find()
+      .sort({ average_rating: -1 })
+      .skip(skipData)
+      .limit(pageSize)
+    
+    if (!books.length) {
+      res.json({ error: 'Oops! Page not found'})
```

Comment:
You should send a status code with this error message. To provide more info to the client, and also to be consistent with your other error messages